org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -64,10 +64,17 @@
 	public static final String PROJECT_CONFIGURATION_IS_NOT_UP_TO_DATE_WITH_POM_XML = "Project configuration is not up-to-date with pom.xml, requires an update.";
 	private final JavaClientConnection connection;
 	private final ProjectsManager projectsManager;
+	private final boolean isDiagnosticTagSupported;
 
+	@Deprecated
 	public WorkspaceDiagnosticsHandler(JavaClientConnection connection, ProjectsManager projectsManager) {
+		this(connection, projectsManager, false);
+	}
+
+	public WorkspaceDiagnosticsHandler(JavaClientConnection connection, ProjectsManager projectsManager, boolean isDiagnosticTagSupported) {
```

Comment:
I'd prefer we passed ClientPreferences or else will need to change the method signature every time there'e an evolution of LSP

Reply:
Back from a trip. This was just addressed in the latest commit.